src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/filter/ChooseSupplierFilter.kt

```diff
@@ -22,16 +22,13 @@ class ChooseSupplierFilter(private val timeSource: TimeSource) : Filter {
 
   @Throws(IOException::class, ServletException::class)
   override fun doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain) {
-    val req = request as HttpServletRequest
-    val res = response as HttpServletResponse
-    val session = req.session
+    val session = (request as HttpServletRequest).session
 
-    when (session.getAttribute("supplier")) {
+    when (session.getAttribute(HtmlController.SUPPLIER_ATTRIBUTE)) {
       null -> {
-        logger.info("no supplier present in the session, redirecting to choose supplier")
+        logger.info("no supplier present in the session, redirecting to '${HtmlController.CHOOSE_SUPPLIER_URL}'")
 
-        session.setAttribute("date", timeSource.date().withDayOfMonth(1))
```

Comment:
One of the key changes here.  Removed this from the filter and put logic into the Controller.